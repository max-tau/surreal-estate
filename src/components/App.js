import React, { useState } from "react";
import "../styles/App.css";
import { Route, Routes, useNavigate } from "react-router-dom";
import { signOut } from "firebase/auth";
import { auth } from "../config/firebase";
import NavBar from "./NavBar";
import Properties from "./Properties";
import AddProperty from "./AddProperty";
import Favourites from "./Favourites";
import CreateAccount from "./CreateAccount";
import SignIn from "./SignIn";

const App = () => {
  const [userId, setUserId] = useState("");
  const handleLogin = (response) => {
    setUserId(response.userID);
  };
  const navigate = useNavigate();
  const handleLogout = () => {
    signOut(auth)
      .then(() => {
        setUserId("");
        navigate("/");
      })
      .catch((err) => console.log(err));
  };

  return (
    <div className="App">
      <NavBar onLogin={handleLogin} onLogout={handleLogout} userId={userId} />
      <Routes>
        <Route path="/" element={<Properties userId={userId} />} />
        <Route path="add-property" element={<AddProperty />} />
        <Route path="favourites" element={<Favourites />} />
        <Route
          path="create-account"
          element={<CreateAccount onSetUser={setUserId} />}
        />
        <Route path="sign-in" element={<SignIn onSetUser={setUserId} />} />
      </Routes>
    </div>
  );
};

export default App;
