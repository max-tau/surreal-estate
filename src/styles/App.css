h2 {
  font-family: "Raleway Dots", cursive;
  font-size: 120px;
  font-weight: 50;
  color: rgb(46, 50, 54);
}
