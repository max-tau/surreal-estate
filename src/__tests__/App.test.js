import { render, screen } from '@testing-library/react';
import App from '../components/App';

test('renders subheading', () => {
  render(<App />);
  const h2Element = screen.getByText(/Surreal Estate/i);
  expect(h2Element).toBeInTheDocument();
});
