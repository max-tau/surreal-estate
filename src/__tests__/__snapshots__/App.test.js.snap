// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders components correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="navbar"
    >
      <img
        alt="Surreal Estate logo"
        class="navbar-logo"
        src="https://mcrcodes.s3.eu-west-2.amazonaws.com/course/surreal-estate/logo.png"
      />
      <ul
        class="navbar-links"
      >
        <li
          class="navbar-links-item"
        >
          <a
            class="navbar-links-item"
            href="/"
          >
            View Properties
          </a>
        </li>
        <li
          class="navbar-links-item"
        >
          <a
            class="navbar-links-item"
            href="/add-property"
          >
            Add a Property
          </a>
        </li>
      </ul>
    </div>
    <div>
      Properties Page
    </div>
  </div>
</DocumentFragment>
`;
